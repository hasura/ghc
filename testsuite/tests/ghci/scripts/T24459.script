:set -XTypeAbstractions

:{
import Data.Kind
import Data.Data


type T6 :: forall k. Type -> Type
type T6 _a = ()

type T5 :: forall a. Type -> Type
type T5 @a _b = Proxy _b

type T4 :: forall a. Type -> Type
type T4 @a _b = Proxy a

type T3 :: forall k. k -> Type
type T3 a = Proxy a

type T2 :: forall k. k -> Type
type T2 @k = Proxy

type T1 :: forall k. k -> Type
type T1 = Proxy

type T0 :: forall k. k -> Type
type T0 = Proxy :: forall k. k -> Type
:}

:i T0
:i T1
:i T2
:i T3
:i T4
:i T5
:i T6

:set -fprint-explicit-kinds

:i T0
:i T1
:i T2
:i T3
:i T4
:i T5
:i T6
