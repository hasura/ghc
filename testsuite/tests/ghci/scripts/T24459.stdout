type T0 :: forall k. k -> *
type T0 = Proxy
  	-- Defined at <interactive>:27:1
type T1 :: forall k. k -> *
type T1 = Proxy
  	-- Defined at <interactive>:24:1
type T2 :: forall k. k -> *
type T2 @k = Proxy
  	-- Defined at <interactive>:21:1
type T3 :: forall k. k -> *
type T3 a = Proxy a
  	-- Defined at <interactive>:18:1
type T4 :: forall {k} (a :: k). * -> *
type T4 @a _b = Proxy a
  	-- Defined at <interactive>:15:1
type T5 :: forall {k} (a :: k). * -> *
type T5 _b = Proxy _b
  	-- Defined at <interactive>:12:1
type T6 :: forall {k} (k1 :: k). * -> *
type T6 _a = ()
  	-- Defined at <interactive>:9:1
type T0 :: forall k. k -> *
type T0 = Proxy
  	-- Defined at <interactive>:27:1
type T1 :: forall k. k -> *
type T1 = Proxy
  	-- Defined at <interactive>:24:1
type T2 :: forall k. k -> *
type T2 @k = Proxy @{k}
  	-- Defined at <interactive>:21:1
type T3 :: forall k. k -> *
type T3 @k a = Proxy @{k} a
  	-- Defined at <interactive>:18:1
type T4 :: forall {k} (a :: k). * -> *
type T4 @{k} @a _b = Proxy @{k} a
  	-- Defined at <interactive>:15:1
type T5 :: forall {k} (a :: k). * -> *
type T5 @{k} @a _b = Proxy @{*} _b
  	-- Defined at <interactive>:12:1
type T6 :: forall {k} (k1 :: k). * -> *
type T6 @{k} @k1 _a = ()
  	-- Defined at <interactive>:9:1
